src/uopool/services/uopool.rs

```diff
@@ -220,22 +220,14 @@ where
                             }
                             Err(error) => {
                                 res.set_result(EstimateUserOperationGasResult::NotEstimated);
-                                res.data =
-                                    serde_json::to_string(&SimulationError::from(match error {
-                                        EntryPointErr::JsonRpcError(err) => {
-                                            SimulateValidationError::<M>::UserOperationExecution {
-                                                message: err.message,
-                                            }
-                                        }
-                                        _ => SimulateValidationError::UnknownError {
-                                            error: format!("{error:?}"),
-                                        },
-                                    }))
-                                    .map_err(|_| {
-                                        tonic::Status::internal(
-                                            "error estimating user operation gas",
-                                        )
-                                    })?;
+                                res.data = serde_json::to_string(&SimulationError::from(
+                                    SimulateValidationError::<M>::UnknownError {
```

Comment:
This should probably be handled differently, the UserOperationExecution is never returned.